scripts/compose-user-provided-service.sh

```diff
@@ -0,0 +1,122 @@
+#!/bin/bash
+set -euo pipefail
+COMPOSE_API_URL='https://api.compose.io/2016-07'
+
+usage() {
+  script=$0
+  cat <<EOF
+  Usage:
+    $script -o <orgname> -t <type>
+
+$script will manually create an instance
+of defined type with the use of compose API and assign it to the specific
+organisation as a CloudFoundry User Provided Service.
+
+Requirements:
+
+ * You must login with the cf client with an "admin" user.
+ * You must have the compose_api_token environment variable set.
+
+Where:
+  -o <orgname> Organisation the service should be assigned to upon creation.
+
+  -t <type>    Service type to be created. Supported:
+               mongodb, redis, postgresql, elastic_search, rethink, rabbitmq,
+               etcd, scylla, mysql, disque
+EOF
+exit 1
+}
+
+get_cluster_data(){
+  compose_cluster_data=$(curl -X GET "${COMPOSE_API_URL}/clusters" \
+    -H "Authorization: Bearer ${COMPOSE_API_TOKEN}" \
+    -H "Content-Type: application/json" \
+    2>/dev/null | jq -r ._embedded.clusters[0])
+
+  compose_cluster_id=$(echo "${compose_cluster_data}" | jq -r .id)
+  compose_account_id=$(echo "${compose_cluster_data}" | jq -r .account_id)
+}
+
+provision_instance() {
+  id=$(uuidgen | cut -c1-13)
+  deployment_name="${TYPE}-${id}"
+
+  local body
+  body=$(printf '{
+    "deployment": {
+      "name": "%s",
+      "account_id": "%s",
+      "cluster_id": "%s",
+      "type": "%s"
+    }
+  }' "${deployment_name}" "${compose_account_id}" "${compose_cluster_id}" "${TYPE}")
+
+  uri=$(curl -X POST "${COMPOSE_API_URL}/deployments" \
+      -H "Authorization: Bearer ${COMPOSE_API_TOKEN}" \
+      -H "Content-Type: application/json; charset=utf-8" \
+      -d "${body}" | jq -r .connection_strings.direct[0])
+
+  pattern='^(([a-z]{3,5})://)?((([^:\/]+)(:([^@\/]*))?@)?([^:\/?]+)(:([0-9]+))?)(\/[^?]*)?(\?[^#]*)?(#.*)?$'
+  [[ "$uri" =~ $pattern ]] || return 1;
+
+  uri=${BASH_REMATCH[0]}
+  uri_user=${BASH_REMATCH[5]}
+  uri_password=${BASH_REMATCH[7]}
+  uri_host=${BASH_REMATCH[8]}
+  uri_port=${BASH_REMATCH[10]}
+}
+
+create_user_provided_service() {
+  local body
+  body=$(printf '{
+    "username": "%s",
+    "password": "%s",
+    "host": "%s",
+    "port": "%s",
+    "uri": "%s"
+  }' "${uri_user}" "${uri_password}" "${uri_host}" "${uri_port}" "${uri}")
+
+  cf create-user-provided-service "${deployment_name}" -p "${body}"
```

Comment:
This hasn't targeted an org/space, so will create this in the current org/space.